### PR TITLE
Implement pagination with an iterator-like API

### PR DESCRIPTION
**Pull request by txsmith**
This adds two properties and one method to the API:
- `canFetchMore`, true iff the last response was paginated and more pages can be fetched.
- `fetchMore()`, call to generate a request for the next page, or for the first page if nor previous request has been made.
- `allResults`, an array containing results from all fetched pages.

Will rebase after #3 has been merged

### CODE REVIEW

File: github-ajax.html [hunk 1]
```diff
@@ -290,26 +398,64 @@
...
           this.accessToken = sharedToken;
         }
 
-        const headers = this.requestHeaders;
-        const request = new Request(this.requestUrl, {
-            method: this.method,
-            headers: headers,
-            body: this._encodeBody(headers)
-          });
+        const headers = this.requestHeaders
```
Comment (eanker) [inline github-ajax.html]: Why remove the `;`

**Comment by eanker** [comment]
I get` 403 forbidden`:

image
 
Note: You have to run my `use-comment-element` on frontend as you have merged the one in code-highlighter. And master now complains about a duplicate comment.html

**Comment by txsmith** [comment]
The access token seems to be missing when `generateRequest()` is called. I'll look into it. Probably some issue with the order of data binding events

**Comment by txsmith** [comment]
I cannot reproduce the problem, does it still error after a `bower install / update`?

**Comment by txsmith** [comment]
Should be fixed now!

**Review by eanker — APPROVED** [review]
Error is solved and works as expected

**Pull request merged**